webapp/channels/src/selectors/guest_tags.test.ts

```diff
@@ -0,0 +1,151 @@
+
+            expect(shouldHideGuestTags(state)).toBe(false);
+        });
+
+        it('should return false when config is null', () => {
+            const state = {
+                entities: {
+                    general: {
+                        config: null,
+                    },
+                },
+            } as unknown as GlobalState;
+
+            expect(shouldHideGuestTags(state)).toBe(false);
+        });
+
+        it('should return false for case-sensitive variations (True, TRUE, etc)', () => {
+            const testCases = ['True', 'TRUE', 'tRuE', 'yes', '1', 'on'];
+
+            testCases.forEach((value) => {
+                const state = {
+                    entities: {
+                        general: {
+                            config: {
+                                HideGuestTags: value,
+                            },
+                        },
+                    },
+                } as unknown as GlobalState;
+
+                expect(shouldHideGuestTags(state)).toBe(false);
+            });
+        });
+
+        it('should return false for empty string', () => {
+            const state = {
+                entities: {
+                    general: {
+                        config: {
+                            HideGuestTags: '',
+                        },
+                    },
+                },
+            } as unknown as GlobalState;
+
+            expect(shouldHideGuestTags(state)).toBe(false);
+        });
+
+        // Memoization test
+        it('should memoize results and return same reference for same input', () => {
+            const state = {
+                entities: {
+                    general: {
+                        config: {
+                            HideGuestTags: 'true',
+                        },
+                    },
+                },
+            } as unknown as GlobalState;
+
+            const result1 = shouldHideGuestTags(state);
+            const result2 = shouldHideGuestTags(state);
+
+            // Should return the same cached result
+            expect(result1).toBe(result2);
+            expect(result1).toBe(true);
+        });
+
+        it('should recompute when config actually changes', () => {
+            const state1 = {
+                entities: {
+                    general: {
+                        config: {
+                            HideGuestTags: 'false',
+                        },
+                    },
+                },
+            } as unknown as GlobalState;
+
+            const state2 = {
+                entities: {
+                    general: {
+                        config: {
+                            HideGuestTags: 'true',
+                        },
+                    },
+                },
+            } as unknown as GlobalState;
+
+            expect(shouldHideGuestTags(state1)).toBe(false);
+            expect(shouldHideGuestTags(state2)).toBe(true);
+        });
```

Comment:
issue: These tests don't really do anything because the value returned by this selector is a primitive and because this selector doesn't do anything complicated enough to memoize